File: app/src/main/java/com/sysu/edu/view/AdapterListener.java
package com.sysu.edu.view;

import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;
import androidx.viewbinding.ViewBinding;

public interface AdapterListener {
    void onBind(RecyclerView.Adapter<RecyclerView.ViewHolder> adapter, RecyclerView.ViewHolder holder, int position);

    void onCreate(RecyclerView.Adapter<RecyclerView.ViewHolder> adapter, ViewBinding binding);

}
interface Adapter2Listener<V extends ViewDataBinding> {
    void onBind(RecyclerView.Adapter<RecyclerViewHolder<V>> adapter, RecyclerViewHolder<V> holder, int position);
    
    void onCreate(RecyclerView.Adapter<RecyclerViewHolder<V>> adapter, V binding);
    
}

File: app/src/main/java/com/miyuyan/sysuer/view/AdapterListener.kt
package com.miyuyan.sysuer.view

import androidx.databinding.ViewDataBinding
import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding

interface AdapterListener {
	fun onBind(adapter: RecyclerView.Adapter<RecyclerView.ViewHolder?>,
	           holder: RecyclerView.ViewHolder,
	           position: Int)
	
	fun onCreate(adapter: RecyclerView.Adapter<RecyclerView.ViewHolder?>, binding: ViewBinding?)
}

internal interface Adapter2Listener<V : ViewDataBinding?> {
	fun onBind(adapter: RecyclerView.Adapter<RecyclerViewHolder<V?>?>?,
	           holder: RecyclerViewHolder<V?>?,
	           position: Int)
	
	fun onCreate(adapter: RecyclerView.Adapter<RecyclerViewHolder<V?>?>?, binding: V?)
}